### PR TITLE
Getcwd

### PR DESCRIPTION
**Pull request by youngfalcon**
Added basic test cases for the getcwd syscall

### CODE REVIEW

**Comment by rennergade** [comment]
Simple and good!

Does chdir work appropriately? I had thought it wasn't working.

**Comment by youngfalcon** [comment]
It seems to work properly in my tests, although I did not go over edge cases on chdir.

If you are referring to our earlier problems with nginx, I suspect that those failures were caused by a broken getcwd as the values passed into chdir might be dependent on it.

File: tests/test_cases/chdir_getcwd.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  char* result = getcwd(cwd, sizeof(cwd));
+
+  if (result == NULL)
+    perror("getcwd() error");
+  else
+    printf("current working directory is: %s :: %s\n", cwd, result);
+  //---------------------------
+  chdir("/test");
+  //---------------------------
```
Comment (kapkic) [inline tests/test_cases/chdir_getcwd.c:16]: Instead of adding two comment lines; `chdir (...);  //Working directory` style could be a bit better, doesn't affect the functionality though.

**Review by kapkic — APPROVED** [review]
LGTM!

**Review by jesings — APPROVED** [review]
These tests look good. I was also of the impression that getcwd did not work correctly because we didn't hook it up from rustposix. Additionally I think the tests could probably have some assertions of strcmps with the expected results of getcwds for further automation but that's not entirely necessary.
